common/achievement/types.ts

```diff
@@ -109,26 +109,23 @@ export type EvaluationResult = {
 export type RelationTypes = 'match' | 'subcourse' | 'global_match' | 'global_subcourse'; // match_all, subcourse_all, all
 
 type ContextLecture = Pick<Lecture, 'start' | 'duration'>;
+export type ContextMatch = {
+    id: number;
+    relation: string | null; // will be null if searching for all matches
```

Comment:
is the relation really null here? What about `global_match` / `global_subcourse`?

`export type RelationTypes = 'match' | 'subcourse' | 'global_match' | 'global_subcourse'; `

Reply:
My idea was that it stores the real relation `match/<id>` in case we need it. In this PR it will be null if we are searching for all matches.